tests/libvmi/lifecycle.go

```diff
@@ -1,46 +1,121 @@
-	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+	vmis.create(loggedInExpecterFactory, virtClient)
+	vmis.waitUntilReady(loggedInExpecterFactory, virtClient)
+}
 
-	By("Waiting until the VMI gets ready")
-	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
+func (vmis VMIPool) create(loggedInExpecterFactory tests.VMIExpecterFactory, virtClient kubecli.KubevirtClient) {
+	for name, vmi := range vmis {
+		By(fmt.Sprintf("Creating VMI %q", name))
+		vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+		Expect(err).NotTo(HaveOccurred(), "VMI %q should be successfully created", name)
+		vmis[name] = vmi
+	}
+}
 
-	return vmi
+func (vmis VMIPool) waitUntilReady(loggedInExpecterFactory tests.VMIExpecterFactory, virtClient kubecli.KubevirtClient) {
+	for name, vmi := range vmis {
+		By(fmt.Sprintf("Waiting until VMI %q gets ready", name))
+		vmis[name] = tests.WaitUntilVMIReady(vmi, loggedInExpecterFactory)
+	}
 }
 
-func CleanupVMI(vmi *v1.VirtualMachineInstance) {
-	if vmi != nil {
-		virtClient, err := kubecli.GetKubevirtClient()
-		Expect(err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
+func (vmis VMIPool) Cleanup() {
+	virtClient, err := kubecli.GetKubevirtClient()
+	Expect(err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
 
-		By("Deleting the VMI")
-		Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})).To(Succeed())
+	errs := []error{}
+	errs = append(errs, vmis.delete(virtClient)...)
+	if err = vmis.waitUntilDisposed(virtClient); err != nil {
+		errs = append(errs, err)
+	}
+	Expect(errs).To(BeEmpty(), "Cleanup of all VMIs should succeed")
+}
+
+func (vmis VMIPool) delete(virtClient kubecli.KubevirtClient) []error {
+	errs := []error{}
+
+	for name, vmi := range vmis {
+		if vmi == nil {
+			continue
+		}
+
+		By(fmt.Sprintf("Deleting VMI %q", name))
+		err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func (vmis VMIPool) waitUntilDisposed(virtClient kubecli.KubevirtClient) error {
+	var runningVMIs []string
+
+	err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		runningVMIs = []string{}
+
+		for name, vmi := range vmis {
+			if vmi == nil {
+				continue
+			}
 
-		By("Waiting for the VMI to be gone")
-		Eventually(func() error {
 			_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.GetName(), &metav1.GetOptions{})
-			return err
-		}, 2*time.Minute, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), "The VMI should be gone within the given timeout")
+			if err == nil {
+				runningVMIs = append(runningVMIs, name)
+			} else if err != nil && !errors.IsNotFound(err) {
+				return false, err
+			}
+		}
+
+		if len(runningVMIs) > 0 {
```

Comment:
if all failed, its an error as well isnt it ?
if so maybe worth to check how many were expected to run versus how many actually managed to run